src/main/java/com/dotloop/loopit/LoopItController.java

```diff
@@ -54,51 +51,102 @@
     @Value("${dotloop.api.endpoint}")
     private String apiEndpoint;
 
-    @RequestMapping("/loopit")
+    private String username = "api user";
+
+    private Request attachHeaders(Request request) {
+        Token token = TokenStore.get(username);
+        request.addHeader("Authorization", "Bearer " + token.getAccessToken())
+                .addHeader("Content-type", "application/json");
+
+        return request;
+    }
+
+    private Request attachBody(Request request,  String bodyData) {
+        request.bodyString(bodyData, ContentType.APPLICATION_JSON);
+
+        return request;
+    }
+
+    private String makeRequest(String type, String params, String bodyData) throws Exception {
+        try {
+            params = params.isEmpty() ? "" : params;
+
+            Request request = attachHeaders(Request.Get(getApiBaseUrl() + params));
+
+            if (type.equalsIgnoreCase("post")) {
```

Comment:
@sweetcoco instead of creating conditional logic on the http method,  a better approach might be to create dedicated method for each request type (ie getRequest, postRequest, ect.).  Since the execution and response processing is the same, you could just have each method use the same shared logic.  That would ensure that the calling code is declaring their request type without having to worry that they are setting the "type" parameter correctly.  Also mirrors the same pattern as the underlying library.